app/controllers/tests_controller.rb

```diff
@@ -2,30 +2,39 @@ class TestsController < ApplicationController
   skip_before_action :check_admin, except: [:index]
 
   def create
-    course = Problem.find(params[:problem_id]).course
+    problem = Problem.includes(assignment: :course).find(params[:problem_id])
+    course = problem.course
     if !current_user_in_course?(course)
       head :forbidden
       return
     end
 
     uploaded_file = params[:file]
 
+    test = Test.new(
```

Comment:
Yup, you did in a previous PR. This is so I can pass the test variable to `render`